SRD/runtextsok.yaml

```diff
@@ -1377,6 +1518,55 @@ components:
           example: da0b20c8-e036-456b-8f0a-6aeb06105f9a
       description: Sökfråga som används för att sätta villkor för textsökning i SRD
 
+    Year:
+      description: Anger årtal för att kunna uttrycka frågor av typen "sök runinskrifter daterade till år 1066".
+      required:
+        - year
+      type: object
+      properties:
+        op:
+          type: string
+          description: Anger logisk operator för användas när flera sökfrågor kombineras.
+            Första frågan i en TextQuery behöver inte ha en logisk operator
+          enum:
+            - AND
```

Comment:
Jag är lite tveksam till OR och AND på Year och TimeInterval om dom ska ligga fristående från varandra och från filter och söksträngar. Det finns väl ingen anledning till att ANDa ihop flera år? Det ska vara år 1220 AND år 1240 blir ju konstigt. Samma med tidsintervallet 1200-1300 AND 1250-1350, då kan man lika gärna skriva 1250-1300 direkt.

Frågan är ju om vi borde blanda alla sorts filtreringar, d v s ha någon slags objekt Group som kan innehålla filter, testsök, årtal, osv plus grupper. Då kan man ju göra arbiträra sökningar som kombinerar ihop vad som helst och så får ni i gränssnittet sätta gränsen för vad man ska få göra. 

Blir kanske inte så kul att implementera dock :)

Reply:
Jo... I princip håller jag med dig. Anledningen till att jag lade den fristående var att jag inte kunde se framför mig hur man skulle göra ett sånt "sökobjekt" och att det skulle blir svårimplementerat. Har du nåt förslag får du gärna hitta på nåt!

Angående att man kan skapa knasiga frågor med t ex år=1200 AND 1300 tycker jag inte är ett problem. Dålig fråga ger dåligt svar.... Dessutom har vi ju redan det problemet i t ex proveniens. Du kan välja Land=Norge och Län=Gotland. Det lär inte bli några bra svar där heller...

Reply:
En bra kommentar om proveniens!

Är det helt enkelt så att det är AND man inte behöver? OR är ju självskriven, typ Gotland OR Öland, och jag kan se NOT också, typ Sverige, NOT Gotland. Iofs kanske AND kan vara bra i det sammanhanget också.